routes.py

```diff
@@ -0,0 +1,663 @@
+from sqlalchemy import or_, and_, inspect, func
+from datetime import datetime, time
+from apscheduler.schedulers.background import BackgroundScheduler
+
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    return render_template('index.html', user=current_user)
+
+
+@app.route('/about', methods=['GET', 'POST'])
+def about():
+    return render_template('about.html', user=current_user)
+
+
+@app.route('/register', methods=['GET', 'POST'])
+def register():
+    # sprawdzam czy current_user jest zalogowany, TRUE przekierowywuje na stronę użytkownika
+    if current_user.is_authenticated:
+        return redirect(url_for('user', username=current_user.username))
+    form = RegistrationForm()
+    if form.validate_on_submit():
+        existing_username = User.query.filter_by(
+            username=form.username.data).first()
+        existing_email = User.query.filter_by(email=form.email.data).first()
+        if existing_username is not None:
+            flash('A account with this username already exists on this server.', 'error')
+        elif existing_email is not None:
+            flash('This email has already been used', 'error')
+        else:
+            user = User(username=form.username.data, email=form.email.data)
+            user.set_password(form.password.data)
+            db.session.add(user)
+            db.session.commit()
+            flash('Congratulations, you are now a registered user!', 'success')
+            login_user(user)
+            return redirect(url_for('user', username=user.username))
+    return render_template('register.html', title='Register', form=form, user=current_user)
+
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    # sprawdzam czy current_user jest zalogowany, TRUE przekierowywuje na stronę użytkownika
+    if current_user.is_authenticated:
+        return redirect(url_for('user', username=current_user.username))
+    user = current_user
+    form = LoginForm()
+    if form.validate_on_submit():
+        user = User.query.filter_by(username=form.username.data).first()
+        if user and user.check_password(form.password.data):
+            login_user(user, remember=form.remember_me.data)
+            next_page = request.args.get('next')
+            if not next_page or url_parse(next_page).netloc != '':
+                next_page = url_for('user', username=user.username)
+            flash('You are logged in', 'success')
+            return redirect(next_page)
+        else:
+            flash('Invalid username or password', 'error')
+    return render_template('login.html', form=form, user=user)
+
+
+# ścieżka strony użytkownika, która pokazuje dane konta, profile oraz pozwala tworzyć profile
+@app.route('/user/<username>', methods=['GET', 'POST'])
+@login_required
+def user(username):
+    user = User.query.filter_by(username=username).first_or_404()
+    profiles = Profile.query.filter_by(user_id=user.id).all()
+    if len(profiles) == 0:
+        profiles = []
```

Comment:
if the profile query returns an empty list, it saves this as an empty list. I know this comes out as redundant, but otherwise the html code does not read it this way (perhaps it reads it as None) and generates a messy view for the empty list.